sdk/src/transaction_builder.rs

```diff
@@ -0,0 +1,309 @@
+        ))
+    }
+
+    pub fn add_recovery_rotation_capability(
+        &self,
+        recovery_address: AccountAddress,
+    ) -> TransactionBuilder {
+        self.transaction_builder(TransactionPayload::Script(
+            scripts::encode_add_recovery_rotation_capability_script(recovery_address),
+        ))
+    }
+
+    pub fn create_child_vasp_account(
+        &self,
+        coin_type: Currency,
+        child_auth_key: AuthenticationKey,
+        add_all_currencies: bool,
+        child_initial_balance: u64,
+    ) -> TransactionBuilder {
+        self.transaction_builder(TransactionPayload::Script(
+            scripts::encode_create_child_vasp_account_script(
+                coin_type.type_tag(),
+                child_auth_key.derived_address(),
+                child_auth_key.prefix().to_vec(),
+                add_all_currencies,
+                child_initial_balance,
+            ),
+        ))
+    }
+
+    pub fn create_recovery_address(&self) -> TransactionBuilder {
+        self.transaction_builder(TransactionPayload::Script(
+            scripts::encode_create_recovery_address_script(),
+        ))
+    }
+
+    pub fn rotate_authentication_key(&self, new_key: AuthenticationKey) -> TransactionBuilder {
+        self.transaction_builder(TransactionPayload::Script(
+            scripts::encode_rotate_authentication_key_script(new_key.to_vec()),
+        ))
+    }
+
+    pub fn rotate_authentication_key_with_recovery_address(
+        &self,
+        recovery_address: AccountAddress,
+        to_recover: AccountAddress,
+        new_key: AuthenticationKey,
+    ) -> TransactionBuilder {
+        self.transaction_builder(TransactionPayload::Script(
+            scripts::encode_rotate_authentication_key_with_recovery_address_script(
+                recovery_address,
+                to_recover,
+                new_key.to_vec(),
+            ),
+        ))
+    }
+
+    pub fn rotate_dual_attestation_info(
+        &self,
+        new_url: Vec<u8>,
+        new_key: Vec<u8>,
+    ) -> TransactionBuilder {
+        self.transaction_builder(TransactionPayload::Script(
+            scripts::encode_rotate_dual_attestation_info_script(new_url, new_key),
+        ))
+    }
+
+    pub fn publish_shared_ed25519_public_key(&self, public_key: Vec<u8>) -> TransactionBuilder {
+        self.transaction_builder(TransactionPayload::Script(
+            scripts::encode_publish_shared_ed25519_public_key_script(public_key),
+        ))
+    }
+
+    pub fn publish_rotate_ed25519_public_key(&self, public_key: Vec<u8>) -> TransactionBuilder {
+        self.transaction_builder(TransactionPayload::Script(
+            scripts::encode_rotate_shared_ed25519_public_key_script(public_key),
+        ))
+    }
+
+    //
+    // Internal Helpers
+    //
+
+    fn transaction_builder(&self, payload: TransactionPayload) -> TransactionBuilder {
+        TransactionBuilder {
+            sender: AccountAddress::ZERO,
+            sequence_number: 0,
```

Comment:
I'm wondering if it would be better to have these as optional fields that can be checked in the `build` method  for the `TransactionBuilder`. I'm not sure if having default values here is preferable over raising an error when calling `build` if these fields aren't set.

Reply:
I'm going to compromise for right now and have these stored as an Option in the builder and then we can panic if the builder doesn't set the sender or seq number since its more about programmer error than something that can actually be fixed at runtime by handling an error.

Reply:
I'm going to compromise for right now and have these stored as an Option in the builder and then we can panic if the builder doesn't set the sender or seq number since its more about programmer error than something that can actually be fixed at runtime by handling an error.